File: frontend/src/model/SchoolPlanForClasses.ts
import {SubjectAndClassesCount} from "./SubjectAndClassesCount";
import {SchoolClass} from "./SchoolClass";

export interface SchoolPlanForClasses {
    id: number | null,
    name: string,
    subjectAndClassesCount: SubjectAndClassesCount[],
    schoolClassesWithTheSchoolPlan: SchoolClass[]
}

File: backend/src/main/kotlin/com/nevexis/backend/schoolManagement/school_plan/SchoolPlanForClasses.kt
package com.nevexis.backend.schoolManagement.school_plan

import com.nevexis.backend.schoolManagement.school_class.SchoolClass
import com.nevexis.backend.schoolManagement.school_schedule.SubjectAndClassesCount

data class SchoolPlanForClasses(
    val id: Int? = null,
    val name: String,
    val subjectAndClassesCount: List<SubjectAndClassesCount>,
    val schoolClassesWithTheSchoolPlan: List<SchoolClass>
)